### PR TITLE
Change mask input of EmbedLayerNormalization op to be optional

### PR DESCRIPTION
**Pull request by tianleiwu**
**Description**: Change mask input of EmbedLayerNormalization op to be optional

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

For GPT-2 model, mask processing is different from Bert model. Here we make the mask input to be optional, so that EmbedLayerNormalization op can be applied to more models.

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/bert/embed_layer_norm.cc [hunk 1]
```diff
@@ -98,13 +98,15 @@ Status EmbedLayerNorm<T>::Compute(OpKernelContext* context) const {
...
-  {
+  if (nullptr != mask) {
     const int* mask_data = mask->template Data<int>();
     for (int b = 0; b < batch_size; b++) {
       mask_index->template MutableData<int>()[b] = static_cast<int>(std::count_if(mask_data + (b * sequence_length),
                                                                                   mask_data + (b * sequence_length) + sequence_length,
                                                                                   [](int v) { return v == 1; }));
     }
+  } else {
+    memset(mask_index->template MutableData<int>(), 0, batch_size * sizeof(int));
```
Comment (yuslepukhin) [inline onnxruntime/contrib_ops/cpu/bert/embed_layer_norm.cc]: >int [](start = 44, length = 3)

int is not a datatype supported in Tensors. It has a different size on different platforms. On WIndows it is 32 bit but on Linux it is 64-bit. Which one would you like to use?

Comment (tianleiwu) [reply onnxruntime/contrib_ops/cpu/bert/embed_layer_norm.cc]: Good catch. I'll update to int32_t.

File: onnxruntime/contrib_ops/cpu/bert/embed_layer_norm.cc [hunk 2]
```diff
@@ -124,13 +124,15 @@ Status EmbedLayerNorm<T>::Compute(OpKernelContext* context) const {
   }
 
   // Calculate mask
-  {
-    const int* mask_data = mask->template Data<int>();
+  if (nullptr != mask) {
```
Comment (fs-eire) [inline onnxruntime/contrib_ops/cpu/bert/embed_layer_norm.cc]: How many outputs are expected when no mask inputted. 1, or 2? if `mask` is null, `Tensor* mask_index = context->Output(1, mask_index_shape);` should not be called

Comment (tianleiwu) [reply onnxruntime/contrib_ops/cpu/bert/embed_layer_norm.cc]: 2.  Input mask is optional, but the output is not, which means we will always have mask_index output. When input mask is NULL, the mask_index output only has dummy result (all zero).

**Pull request merged**